### ISSUE TITLE
URL with '?' do not parse correctly

### ISSUE THREAD

**Issue by anoadragon453** [Antox/Antox#314]
For instance, sending someone this URL: https://download.cyanogenmod.org?device=t0lte

On qTox:
![screenshot_2016-07-12_00-16-59](https://cloud.githubusercontent.com/assets/1342360/16758379/fa80103e-47c5-11e6-9b46-0dbbee7f5cbc.png)

On Antox:
![screenshot_20160712-001208](https://cloud.githubusercontent.com/assets/1342360/16758314/ae518e04-47c5-11e6-8af5-199456ab3d5e.png)

[Here is qTox's link parsing code for reference](https://github.com/tux3/qTox/blob/master/src/chatlog/chatmessage.cpp#L271)

**Comment by wiiaam** [comment]
Issue has been raised here http://stackoverflow.com/questions/36328675/android-androidautolink-web-does-not-support-query-parameters-after-domains

I'll have a look and see if I can add a custom regex to include them.

+1 on that font btw

**Comment by anoadragon453** [comment]
qTox does it entirely with regex so it shouldn't be too bad as long as it doesn't depend too much on Qt libraries.

And thanks, Ubuntu Light Mono, simple but effective.

**Comment by anon8675309** [comment]
If someone points me to the Antox code which needs looked at, I'll take a stab at this one.  That stack overflow link hit the nail on the head.  The forward slash is required by RFC-1738, so it is technically correct to not consider http://www.test.com?p1=v1&p2=v2 a link (which means that java.net.URL doesn't conform to the RFC, which is handy but also a little disconcerting).

**Comment by wiiaam** [comment]
https://github.com/Antox/Antox/blob/master/app/src/main/res/layout/chat_message_row_text.xml#L49

Currently this is what autolinks messages, basically you would have to remove this then write your own regex for matching urls and linking them

**Comment by anon8675309** [comment]
Valid URLs are handled correctly (see attached screenshot).  Per my last comment, RFC-1738 requires a / before a ?  It looks like we're running into one of those "this isn't a link, but others consider it a link the software should violate the RFC and consider it a link anyway" problems.  Google is following the RFC with the autoLink feature of a TextView, qTox apparently does not.  Who is right?  It's somewhat subjective.  RFCs are supposed to be the authority and anyone who doesn't confirm is doing it wrong, by definition.  On the other hand, it's something that obviously is something users want.

If Antox is to be changed, the change should be implemented in TextView, which, as far as I know, is maintained by Google. So if the consensus is that this should be changed, then we should raise the issue to them so it can be fixed across all Android apps, not just Antox.

![screenshot_20170219-185750](https://cloud.githubusercontent.com/assets/4683783/23108614/870bcdfe-f6d5-11e6-9e55-38da5075c468.png)

**Comment by wiiaam** [comment]
Yeah I'm curious to see what other applications have parsing support for these kinds of urls

**Comment by anon8675309** [comment]
I looked into this a little more and realized that RFC-1738 (URLs) has been superseded by RFC-2396 (URIs), however that also dictates that there will not be a query (the section after a question mark) without a path (the section starting with a slash).  So the conclusion remains the same: What Google is doing is right.

One option to fix this would be to modify the message on the way out to fix the URL, but that means changing what the person sends and whether or not that is a good idea is subjective.  I'd argue that if this is the way we want to go, it should be done on the sending side, not the receiving side.  I realize it's just adding a slash, but I feel the message contents should be consistent on both the sender and recipient's screen.  I also feel strongly that the user should have a choice in the matter, however I don't object to opting out rather than having to opt in.  All of this isn't going to fix the qTox -> Antox use case which started this issue, but we could address the Antox -> Antox use case and/or raise the question with the qTox developers...

**Comment by zoff99** [comment]
the exmaple is just not a valid URL
